### DISCUSSION TITLE
Remote buttons (door switches)

### DISCUSSION THREAD

**Discussion by EvilHomerGD** [theyosh/TerrariumPI#751] [Ideas]
@theyosh what are the chances of adding some additional remote (JSON) functionalities? A little background for you: I have multiple enclosures setup throughout my house (or is it a zoo?). I've been looking at ways to add those remote sensors and come up with a fairly simple solution. I have setup a Pi Pico W to spit out sensor information (temp and humidity) to a JSON page. I've been toying with a reed switch for the doors and would love to be able to read that state into my main TerrariumPi interface.

**Comment by theyosh** [comment]
Hi,

yeah this is possible, but the door sensors are normally checked multiple times a second in order to get a realtime action. So, with remote doors, that has to be lowered. So, I can add also remote buttons, with an extra option of how fast I can check it. Like 1 time per X seconds.

And JSON is quite simple to use, so that is possible. But it will behave a bit slower due to the amount of requests we can make per X seconds.

**Reply by EvilHomerGD** [reply]
That would be awesome given the difficulties in getting new pi 4s. I gave writing the code a go myself but I need to spend a lot more time reviewing the code to get the level of knowledge that you have already. All of my efforts kept returning a door open condition so I think I missed something. I'm still teaching myself python, so that really isn't a surprise.

**Reply by theyosh** [reply]
Made a working version in branch 4.x.y.z And please continue on the issue at #752

**Reply by EvilHomerGD** [reply]
I did  a checkout of the branch and then ran through the install script again just to make sure I wasn't missing any dependencies but I'm getting a repeated error in the logs. I posted the message to the issue notes.

**Comment by theyosh** [comment]
I made an issue.. so we can continue there with this. https://github.com/theyosh/TerrariumPI/issues/752
